Shared/Flows/Onboarding/OnboardingViewController.swift

```diff
@@ -317,22 +311,35 @@ class OnboardingViewController: SwitchableContentViewController<OnboardingConten
                 let reservation = try await Reservation.getObject(with: reservationId)
                 self.reservationId = reservationId
                 if let userId = reservation.createdBy?.objectId {
-                    try await self.updateReservationCreator(with: userId)
+                    try await self.updateInvitor(with: userId)
                     await self.hideLoading()
                     self.current = .phone(self.phoneVC)
                 }
 
             }
         case .pass(passId: let passId):
-            break
+            Task {
+                let pass = try await Pass.getObject(with: passId)
+                self.passId = passId
+                if let userId = pass.owner?.objectId {
+                    try await self.updateInvitor(with: userId)
+                    await self.hideLoading()
+                    self.current = .phone(self.phoneVC)
+                }
+            }
         }
     }
 
-    private func handleNameSuccess() {
+    private func handleNameSuccess(for name: String) {
+        self.fullName = name
         // User has been allowed to continue
         if User.current()?.status == .inactive {
             #if APPCLIP
-            self.current = .waitlist(self.waitlistVC)
+            Task {
```

Comment:
If for some reason we change the flow, we can save the name before we activate the user.